init.c

```diff
@@ -83,14 +84,22 @@ do_init(void)
 	}
 
 	/* get system identifier of the current database.*/
-	controlFile = get_controlfile(pgdata, &crc_ok);
-
-	if (!crc_ok)
-		ereport(WARNING,
-				(errmsg("control file appears to be corrupt"),
-				 errdetail("Calculated CRC checksum does not match value stored in file.")));
-	sysid = controlFile->system_identifier;
-	pg_free(controlFile);
+	snprintf(controlFilePath, MAXPGPATH, "%s/global/pg_control", pgdata);
+	if (fileExists(controlFilePath))
+	{
+		controlFile = get_controlfile(pgdata, &crc_ok);
+		if (!crc_ok)
+			ereport(WARNING,
+					(errmsg("control file appears to be corrupt"),
+					 errdetail("Calculated CRC checksum does not match value stored in file.")));
+		sysid = controlFile->system_identifier;
+		pg_free(controlFile);
+	}
+	else
+	{
+		elog(ERROR, _("pg_controldata file \"%s\" does not exist"),
```

Comment:
Perhaps the following words should be added
Make sure the path to the data cluster directory is correct.

Reply:
Thanks, I added errhint.